rubi/docs/examples/example_callback.py

```diff
@@ -0,0 +1,143 @@
+    """
+    raw_event = event_type(block_number=event_data["blockNumber"], **event_data["args"])
+    print('testing callback:')
+    print(raw_event)
+
+def compare_eth_offer(pair_name: str, event_type: Type[BaseEvent], event_data: EventData, base_asset: ERC20, quote_asset: ERC20) -> None:
+    """This is a simple callback function that checks if the offer is an ETH offer, and if so get the current ETH price from Coinbase."""
+
+    # handle the raw event
+    raw_event = event_type(block_number=event_data["blockNumber"], **event_data["args"])
+
+    # we know this type is an EmitOfferEvent, so we can cast it
+    if raw_event.buy_gem == base_asset.address:
+        print('this is a buy offer (buying WETH with USDC)')
+        weth_amt = raw_event.buy_amt
+        usdc_amt = raw_event.pay_amt
+        weth_amt_formatted = base_asset.to_decimal(weth_amt)
+        usdc_amt_formatted = quote_asset.to_decimal(usdc_amt)
+        print(f'an offer to buy {weth_amt_formatted} WETH with {usdc_amt_formatted} USDC at a price of {Decimal(usdc_amt_formatted/weth_amt_formatted)} USDC/WETH')
+
+    elif raw_event.pay_gem == base_asset.address:
+        print('this is a sell offer (selling WETH for USDC)')
+        weth_amt = raw_event.pay_amt
+        usdc_amt = raw_event.buy_amt
+        weth_amt_formatted = base_asset.to_decimal(weth_amt)
+        usdc_amt_formatted = quote_asset.to_decimal(usdc_amt)
+        print(f'an offer to sell {weth_amt_formatted} WETH for {usdc_amt_formatted} USDC at a price of {Decimal(usdc_amt_formatted/weth_amt_formatted)} USDC/WETH')
+    
+    # check if the base asset is ETH
+    if base_asset.symbol == "WETH":
+
+        # get the current ETH price 
+        response = requests.get('https://api.coinbase.com/v2/prices/ETH-USD/spot')
+        data = response.json()
+
+        if response.status_code == 200:
+            price = data['data']['amount']
+            print(f'the current ETH price on coinbase is: {price}')
+        else:
+            return "Error: Unable to retrieve price"
+
+# create a queue to receive messages
+queue = Queue()
+
+# create client
+client = Client.from_http_node_url(
+    http_node_url=http_node_url,
+    custom_token_addresses_file="custom_token_addresses.yaml",
+    wallet=wallet,
+    key=key,
+    message_queue=queue
+)
+
+# initialize ERC20 clients for the base and quote assets
+base_asset = ERC20.from_network(name='WETH', network=client.network)
+quote_asset = ERC20.from_network(name='USDC', network=client.network)
+
+# add the WETH/USDC pair to the client - optional allowances can be set here
+client.add_pair(pair_name="WETH/USDC") # , base_asset_allowance=Decimal("1"), quote_asset_allowance=Decimal("2000")
+
+# start listening to offer events created by your wallet on the WETH/USDC market and the WETH/USDC orderbook
+client.start_event_poller("WETH/USDC", event_type=EmitOfferEvent, event_handler=partial(compare_eth_offer, base_asset=base_asset, quote_asset=quote_asset))
```

Comment:
While this does work I think it is more complicated for users and shouldn't be advertised as the default path

Reply:
i agree, i think the approach shown in #35 and more broadly in #34 is a much better approach, separates out the event handling logic to be based off the queue instead of at the `event_poller` level. lets definitely point user's to this pattern as the default, and we can simply have the `event_handler` buried a bit deeper with a clear breakdown of required structure for users who still choose to go this route.